core/src/main/java/hudson/WebAppMain.java

```diff
@@ -396,8 +396,10 @@ public void contextDestroyed(ServletContextEvent event) {
         if(instance!=null)
             instance.cleanUp();
         Thread t = initThread;
-        if (t!=null)
+        if (t != null && t.isAlive()) {
+            LOGGER.log(Level.INFO, "Shutting down a Jenkins instance that was still starting up", new Throwable("reason"));
```

Comment:
This case does not indicate an error, so `WARNING` is inappropriate, but a stack trace is the only way to diagnose _who_ caused the thread to be interrupted and _why_. This was critical missing information in a recent customer problem.